Frameworks/Foundation/NSURLProtocol_WinHTTP.mm

```diff
@@ -264,7 +283,20 @@ - (void)startLoading {
         NSError* error = nil;
 
         try {
-            auto httpRequestMessage = _requestMessageForNS(self.request);
+            auto httpRequestMessage = _requestMessageForNS(self.request, _flattenedBodyStream);
+
+            // VSO 6693048: If this properly supports streaming bodies instead of flattening the data up front,
+            // more accurate callbacks can be done
+            NSInteger contentSize = self.request.HTTPBody ? self.request.HTTPBody.length : _flattenedBodyStream.size();
```

Comment:
Nah, the downstream delegates want NSInteger https://developer.apple.com/reference/foundation/nsurlconnectiondatadelegate/1418264-connection?language=objc